packages/astro/src/core/build/static-build.ts

```diff
@@ -418,27 +418,26 @@ async function ssrMoveAssets(opts: StaticBuildOptions) {
 }
 
 /**
- * This function takes as input the virtual module name of an astro page and transform
- * to generate an `.mjs` file:
+ * This function takes the virtual module name of any page entrypoint and
+ * transforms it to generate a final `.mjs` output file.
  *
  * Input: `@astro-page:src/pages/index@_@astro`
- *
  * Output: `pages/index.astro.mjs`
+ * Input: `@astro-page:../node_modules/my-dep/injected.astro@_@astro`
+ * Output: `pages/injected.mjs`
  *
- * 1. We remove the module id prefix, `@astro-page:`
- * 2. We remove `src/`
- * 3. We replace square brackets with underscore, for example `[slug]`
- * 4. At last, we replace the extension pattern with a simple dot
- * 5. We append the `.mjs` string, so the file will always be a JS file
+ * 1. We clean the `facadeModuleId` by removing the `@astro-page:` prefix and `@_@astro` suffix
```

Comment:
That's incorrect, what we replace is `@_@`, which was used to hide the file extension: index.astro -> index@_@astro -> index.astro